Automattic/wp-calypso
# A4A: Add "Next Steps" Onboarding Tours

Resolves https://github.com/Automattic/jetpack-genesis/issues/292

## Proposed Changes

* Adds the guided tour components to the Sites screen.
  * The tour will be shown when the page's query string includes `tour=sites-walkthrough` or `tour=add-new-site`. These are linked in the Overview page's "Next Steps" card.
  * You can click step by step through each popover.
  * If you open a site details pane half way through a tour, the popover component will be hidden automatically. Closing the site will resume the tour.
  * Uses React Context to provide data about the current tour to all `GuidedTourStep` components. 
  * Uses individual `GuidedTourStep` components to ensure each step is only rendered when it's anchor is available.

## Testing Instructions



* From the A4A overview page, click the "get familiar with the sites management dashboard" and "learn how to add new sites" steps to start each tour.
* Verify you are redirected to the sites screen with a `tour` parameter in the query string.
* Click through the tour and verify each step renders appropriately.
* Verify skipping the tour dismisses all steps.
* Verify a dismissed tour does not re-start on page refresh.

## Pre-merge Checklist




## Screenshots

<img width="570" alt="Screenshot 2024-04-01 at 9 15 20 PM" src="https://github.com/Automattic/wp-calypso/assets/10933065/80f02396-1368-4da9-b091-cb7066aebdda">
